labs-ios-starter/Surveys/View Controllers/TopicQuestionsViewController.swift

```diff
@@ -27,31 +115,41 @@ class TopicQuestionsViewController: UIViewController {
 
 extension TopicQuestionsViewController: UICollectionViewDataSource, UICollectionViewDelegate {
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return questions + 1
+        return questions.count
     }
     
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
         switch indexPath.row {
-            case 0..<(questions):
-                let cell = questionsCollectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath)
-                cell.setDimensions(width: view.frame.width - 40, height: 40)
+        case 0..<(questions.count):
+                let cell = questionsCollectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath) as! QuestionCollectionViewCell
+
+                cell.updateViews( adjustingForWidthOf: view,
+                                  question: questions[indexPath.item].question )
                 return cell
+
             default:
-                let cell = questionsCollectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier2, for: indexPath)
-                cell.setDimensions(width: view.frame.width - 40, height: 35)
+
+                let cell = questionsCollectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier2, for: indexPath) as! QuestionCollectionViewCell
+                //TODO: We need a question/text to pass in
+                cell.updateViews(adjustingForWidthOf: view)
+
                 return cell
         }
     }
     
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        guard indexPath.row == questions else { return }
-        
-        questions+=1
+        guard indexPath.row == questions.count else { return }
+
+        //not sure what to do with this
+        //questions+=1
```

Comment:
This is saying, ignore all the cells being tapped. BUT if it was the special cell at `questions.count + 1`, that means the user tapped that because they want to add a new cell. So incrementing `questions+=1` and reloading will repopulate the collection with a new question cell and a new last cell to add another.

Reply:
We can keep all this logic for dynamic number of cell questions commented out for now, I was definitely rushed in putting this flow together in a short time frame. This all can be done better later.